chain/src/chain.rs

```diff
@@ -142,7 +141,7 @@ impl OrphanBlockPool {
 /// maintains locking for the pipeline to avoid conflicting processing.
 pub struct Chain {
 	db_root: String,
-	store: Arc<store::ChainStore>,
+	store: Arc<RwLock<store::ChainStore>>,
```

Comment:
I'm fine with the use of a `RwLock` but can't this be pushed down into our LMDB store? All regular operations would be considered a read, except for the close/open of the DB which would take the write.

Reply:
Took a bit of doing and testing, but think I've managed it in the latest push.